src/test/ui/impl-trait/equality.stderr

```diff
@@ -1,6 +1,9 @@
 error[E0308]: mismatched types
   --> $DIR/equality.rs:25:5
    |
+21 | fn two(x: bool) -> impl Foo {
+   |                    -------- expected `_` because of return type
```

Comment:
This note seems odd;  presumably we expected i32? I'm not sure, but that seems to be the case here...